### PR TITLE
Fixes #3984: Allow for Redis Sentinel Connection Configuration

### LINKED ISSUE #3984: Allow Redis Connections to Use Sentinel
### Environment
* Python version:  3.7.3
* NetBox version:  2.7.2

### Proposed Functionality

My proposal is to allow `SENTINELS` and `SERVICE` as optional keys to the `REDIS` dictionaries, like so. If these are both provided and non-empty, then the application would attempt to use the Redis Sentinel connection information. Note that the current way of configuring normal Redis would not change.

```python
# Using the current method 
REDIS = {
    'webhooks': {
        'HOST': 'webhooks_redis',
        'PORT': 6379,
        'PASSWORD': '',
        'DATABASE': 0,
        'DEFAULT_TIMEOUT': 300,
        'SSL': False,
    },
    'caching': {
        'HOST': 'cache_redis',
        'PORT': 6379,
        'PASSWORD': '',
        'DATABASE': 0,
        'DEFAULT_TIMEOUT': 300,
        'SSL': False,
    }
}

# Using a Sentinel instance
REDIS = {
    'webhooks': {
        'SENTINELS': (
            ('sentinel.redis.example.com', 1234),  # Each is defined as (HOST, PORT) tuple
        ),
        'SERVICE': 'webhooks_redis',                   # Sentinel service to connect to
        'PASSWORD': '',
        'DATABASE': 0,
        'DEFAULT_TIMEOUT': 300,
        'SSL': False,
    },
    'caching': {
        'SENTINELS': (
            ('sentinel.redis.example.com', 1234),  # Each is defined as (HOST, PORT) tuple
        ),
        'SERVICE': 'webhooks_redis',                   # Sentinel service to connect to
        'PASSWORD': '',
        'DATABASE': 0,
        'DEFAULT_TIMEOUT': 300,
        'SSL': False,
    }
}
```

### Use Case

It was nice that the webhooks and cache Redis connection settings were separated, but it would be ever nicer to be able to configure one (or both) to use Redis Sentinel connections for the sake of High Availability. Not only may this be preferred, but in some environments, it might be necessary due to the availability of developer environments.

### Database Changes
None.

### External Dependencies
None.

### PR DESCRIPTION
**Pull request by dstarner**
### Fixes: #3984 

Adds functionality and documentation to allow Redis for both the webhook and caching layers to use [Redis Sentinel](https://redis.io/topics/sentinel).

### CODE REVIEW

**Comment by DanSheps** [comment]
One quick comment, we should perhaps not have the docs under requirements.md.  Maybe in a separate sentinel configuration doc or something (will wait for @jeremystretch for this), I just think if we include this in requirements it will be viewed as a requirement.

**Comment by dstarner** [comment]
I thought about the location for the docs a little, and I decided to place them under the initial Redis docs since 1) using sentinel for redis does not change the fact that it is required and 2) it reduces searching around the documentation if someone needs to use Sentinel from the get-go

I will fix the linting error (as well as need to update the connection check in the `extras.apps.AppConfig`).

**Comment by jeremystretch** [comment]
Hold off on this until #4159 is fixed please as it will introduce a conflict.

File: netbox/netbox/settings.py [hunk 1]
```diff
@@ -394,28 +407,35 @@ def _setting(name, default=None):
 #
 # Caching
 #
-
-if CACHING_REDIS_SSL:
-    REDIS_CACHE_CON_STRING = 'rediss://'
+if CACHING_REDIS_USING_SENTINEL:
+    CACHEOPS_SENTINEL = {
+        'locations': CACHING_REDIS_SENTINELS,
```
Comment (DanSheps) [inline netbox/netbox/settings.py]: This line appears to be throwing the PEP8 error

**Review by DanSheps — CHANGES_REQUESTED** [review]
Could you also fix up the PEP8 error?

**Comment by dstarner** [comment]
Oops, I missed you comment about holding off...I will remove that last commit and wait for the merge of #4159. I'll fix the linting errors once I rebase and remove any conflicts

**Comment by dstarner** [comment]
Alright @jeremystretch this should be good to go now, thanks for getting the other one in so quickly

**Comment by jeremystretch** [comment]
I've asked @lampwins to sanity-check this as well as he's probably most familiar with the caching configuration.

**Comment by dstarner** [comment]
That's fine. For more context, I am currently running with the same settings changes in our production instance(s) and we haven't had any problems

**Comment by DanSheps** [comment]
In addition to.that, I do think the docs for this particular piece should go in the optional settings doc file

**Comment by dstarner** [comment]
Yea, I mean, it doesn't matter too much to me as long as its *somewhere*, so I don't mind moving it. I'll update it :)

**Comment by DanSheps** [comment]
We can leave it as is, I was just worried some people might see it and think it is something that is required

**Comment by dstarner** [comment]
I guess I'll leave it as is unless anyone has a strong opinion one way or another.

**Pull request merged**